packages/headless/src/modules/global-search/services/elastic.service.ts

```diff
@@ -0,0 +1,178 @@
+import { HttpStatus, Injectable } from "@nestjs/common";
+import { ElasticsearchService } from "@nestjs/elasticsearch";
+import { searchConfig } from "config/search";
+import { IProductSearchResponse } from "models";
+import { ProductSrarchDatabase } from "src/database/mongodb/search";
+import { Product } from "src/entity/product";
+import { errorResponse, successResponse } from "src/utils/response";
+import { IServiceResponse } from "src/utils/response/service.response.interface"; 
+import { ISearchProductResponse } from "../rest/dto";
+import { productSearchSchema } from "./product.schema";
+
+@Injectable()
+export class ElasticService {
+  constructor(private readonly esService: ElasticsearchService, private readonly productSearchDB: ProductSrarchDatabase) {}
+
+
+  public async createIndex() {
+    try {
+      const index = searchConfig.index
+      await this.esService.indices.delete({ index });
+      await this.esService.indices.create({ index });
+    } catch (error) {
+      console.log(error)
+    }
+    
+  }
+
+  private async createSchema(){
+    await this.esService.indices.putMapping({ 
+      index: searchConfig.index, 
+      type: searchConfig.index,
+      include_type_name: true,
+      body: { 
+          properties: productSearchSchema
+      }
+  });
+  }
+  
+  private mapSearchData(e: Product): Record<string, any> { 
+      let data: any = {} 
+
+      data.infoProductId = e.id || ''
+      data.infoName = e?.info?.name || ''
+      data.infoShortDescription = e?.info?.shortDescription || ''
+      data.infoFullDescription = e?.info?.fullDescription || ''
+      data.infoSku = e?.info?.sku || ''
+      data.infoPrice = e?.info?.price || 0
+
+      data.metaKeywords = e?.meta?.keywords || []
+      data.metaTitle = e?.meta?.title || ''
+      data.metaDescription = e?.meta?.description || ''
+
+      data.brands = e?.brands || []
+      data.categories = e?.categories || []
+      data.manufacturer = e?.manufacturer  || {}
+      data.photos = e?.photos.map(p => p.url)  || []
+      data.tags = e?.tags || []
+
+      return data;
+  } 
+
+  async getSingleByElasticId(id: string): Promise<any>{
+    const {body} =  await this.esService.get({index: searchConfig.index, id})
+    return body._source
+  }
+
+  async singleInsert(productData: Product): Promise<number> { 
```

Comment:
where you are using this function?

Reply:
This will be usefull when admin needs to update search database using apis.
Or when a new product need to add on search database in future.